File: apps/api-rs/src/models/conversation.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

use super::ConversationStatus;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConversationSession {
    pub id: Uuid,
    pub tenant_id: Uuid,
    pub user_id: Uuid,
    pub title: String,
    pub kb_ids: Vec<Uuid>,
    pub status: ConversationStatus,
    pub summary: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateConversationRequest {
    pub kb_ids: Vec<Uuid>,
    #[serde(default)]
    pub title: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConversationListItem {
    pub conversation_id: Uuid,
    pub title: String,
    pub last_message_preview: Option<String>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConversationListResponse {
    pub items: Vec<ConversationListItem>,
    pub next_cursor: Option<String>,
}

File: apps/api-bun/src/models/conversation.ts
// 移植自 apps/api-rs/src/models/conversation.rs
import type { ConversationStatus } from './index.ts';

export interface ConversationSession {
  id: string; tenant_id: string; user_id: string; title: string; kb_ids: string[];
  status: ConversationStatus; summary: string | null;
  created_at: string; updated_at: string;
}
export interface CreateConversationRequest {
  kb_ids: string[]; title?: string | null;
}
export interface ConversationListItem {
  conversation_id: string; title: string; last_message_preview: string | null; updated_at: string;
}
export interface ConversationListResponse {
  items: ConversationListItem[]; next_cursor: string | null;
}
